systems/subscriber/subscriber-registry/pkg/server/subscriber.go

```diff
@@ -0,0 +1,288 @@
+	"github.com/ukama/ukama/systems/subscriber/subscriber-registry/pkg"
+	"github.com/ukama/ukama/systems/subscriber/subscriber-registry/pkg/db"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type SubcriberServer struct {
+	subscriberRepo       db.SubscriberRepo
+	msgbus               mb.MsgBusServiceClient
+	subscriberRoutingKey msgbus.RoutingKeyBuilder
+	pb.UnimplementedSubscriberRegistryServiceServer
+}
+
+
+
+func NewSubscriberServer(subscriberRepo db.SubscriberRepo, msgBus mb.MsgBusServiceClient) *SubcriberServer {
+	return &SubcriberServer{subscriberRepo: subscriberRepo,
+		msgbus:         msgBus,
+		subscriberRoutingKey: msgbus.NewRoutingKeyBuilder().SetCloudSource().SetContainer(pkg.ServiceName)}
+}
+
+
+
+func (s *SubcriberServer) Add(ctx context.Context, req *pb.AddSubscriberRequest) (*pb.AddSubscriberResponse, error) {
+	logrus.Infof("Adding subscriber: %v", req)
+	networkID_uuid := uuid.FromStringOrNil(req.GetNetworkID())
+	orgID_uuid := uuid.FromStringOrNil(req.GetOrgID())
+
+	subscriberID_uuid, err := uuid.NewV4()
+	if err != nil {
+		logrus.Errorf("Failed to generate UUID: %s", err)
+		return nil, err
+	}
+
+	timestamp := &timestamppb.Timestamp{
+		Seconds: req.DateOfBirth.GetSeconds(),
+		Nanos:   req.DateOfBirth.GetNanos(),
+	}
+
+	birthday := timestamp.AsTime()
+
+	subscriber := &db.Subscriber{
+		OrgID:orgID_uuid,
+		SubscriberID:          subscriberID_uuid,
+		FirstName:             req.GetFirstName(),
+		LastName:              req.GetLastName(),
+		NetworkID:             networkID_uuid,
+		Email:                 req.GetEmail(),
+		PhoneNumber:           req.GetPhoneNumber(),
+		Gender:                req.GetGender(),
+		Address:               req.GetAddress(),
+		ProofOfIdentification: req.GetProofOfIdentification(),
+		DOB:                   birthday,
+		IdSerial:              req.GetIdSerial(),
+	}
+	err = s.subscriberRepo.Add(subscriber)
+	if err != nil {
+		logrus.Error("error while adding subscriber" + err.Error())
+		return nil, grpc.SqlErrorToGrpc(err, "subscriber")
+	}
+
+	return &pb.AddSubscriberResponse{
+		Subscriber: &pb.Subscriber{
+			OrgID: orgID_uuid.String(),
+			SubscriberID:          subscriberID_uuid.String(),
+			FirstName:             req.GetFirstName(),
+			LastName:              req.GetLastName(),
+			NetworkID:             networkID_uuid.String(),
+			Email:                 req.GetEmail(),
+			PhoneNumber:           req.GetPhoneNumber(),
+			Gender:                req.GetGender(),
+			Address:               req.GetAddress(),
+			ProofOfIdentification: req.GetProofOfIdentification(),
+			DateOfBirth:           birthday.GoString(),
+			IdSerial:              req.GetIdSerial()},
+	}, nil
+
+}
+
+func (s *SubcriberServer) Delete(ctx context.Context, req *pb.DeleteSubscriberRequest) (*pb.DeleteSubscriberResponse, error) {
+	subscriberID := req.GetSubscriberID()
```

Comment:
doing later on `	subscriberIDUUID, err := uuid.FromString(subscriberID)`  makes this line useless.